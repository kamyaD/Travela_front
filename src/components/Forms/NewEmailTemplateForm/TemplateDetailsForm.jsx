import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import  InputRenderer from '../FormsAPI';
import FormContext from '../FormsAPI/FormContext/FormContext';
import formMetadata from '../FormsMetadata/TemplateDetailsMetadata/TemplateDetailsLabels';
import './NewEmailTemplateForm.scss';
import SubmitArea from './FormFieldSets/SubmitArea';

class TemplateDetailsForm extends PureComponent {

  onCancel = (event) => {
    event.preventDefault();
    const { closeModal } = this.props;
    closeModal();
  };

  render() {

    const { renderInput } = new InputRenderer({inputLabels: formMetadata });
    const { selectedTemplate } = this.props;
    return (
      <div>

        <FormContext
          targetForm={this}
          validatorName="myCustomValidator" // uses FormAPI's default validator if this is undefined
        >
          <form className="template-details">
            <fieldset>
              <div className="input-group template-details">
                {renderInput('subject', 'text', {className: 'subject-input',
                  disabled: true, value: selectedTemplate.subject })}
                {renderInput('message', 'textarea', {className: 'message-input',
                  disabled: true, value: selectedTemplate.message})}
              </div>
              <hr />
            </fieldset>
            <SubmitArea
              onCancel={this.onCancel}
              send="Edit template"
              reversed
            />
          </form>
        </FormContext>
      </div>
    );
  }
}

TemplateDetailsForm.propTypes = {
  selectedTemplate: PropTypes.object,
  closeModal: PropTypes.func
};

TemplateDetailsForm.defaultProps = {
  selectedTemplate: {},
  closeModal: () => {}
};


export default TemplateDetailsForm;
