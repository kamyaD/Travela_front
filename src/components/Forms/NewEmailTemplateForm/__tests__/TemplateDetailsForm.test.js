import React from 'react';
import { mount } from 'enzyme';
import TemplateDetailsForm from '../TemplateDetailsForm';

describe('<TemplateDetailsForm />', () => {
  const props = { 
    selectedTemplate: {}, 
    closeModal: jest.fn(),
  };

  const event = {
    target: {
      preventDefault: jest.fn()
    }
  };

  it('renders without crashing', () => {
    const wrapper = mount(<TemplateDetailsForm {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('calls oncancel function', () => {
    const wrapper = mount(<TemplateDetailsForm {...props} />);
    wrapper.find('button#cancel').simulate('click');
    expect(props.closeModal).toHaveBeenCalled();
  });
  
});
