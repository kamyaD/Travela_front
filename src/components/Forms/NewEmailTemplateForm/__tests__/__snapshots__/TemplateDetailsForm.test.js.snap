// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TemplateDetailsForm /> renders without crashing 1`] = `
<div>
  <form
    className="template-details"
  >
    <fieldset>
      <div
        className="input-group template-details"
      >
        <div
          className="form-input subject-input "
        >
          <label
            htmlFor="subject"
          >
            Subject
            <span
              style={
                Object {
                  "color": "red",
                }
              }
            >
               * 
            </span>
          </label>
          <input
            autoComplete="off"
            className=" subject-input"
            disabled={true}
            label="Subject"
            name="subject"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
          />
          <span
            className="error"
          />
        </div>
        <div
          className="form-input message-input "
        >
          <label
            htmlFor="message"
          >
            Message
            <span
              style={
                Object {
                  "color": "red",
                }
              }
            >
               * 
            </span>
          </label>
          <textarea
            autoComplete="off"
            className=" textarea-box message-input"
            disabled={true}
            label="Message"
            name="message"
            onBlur={[Function]}
            onChange={[Function]}
            rows="10"
            type="textarea"
          />
          <span
            className="error"
          />
        </div>
      </div>
      <hr />
    </fieldset>
    <fieldset>
      <div
        className="submit-area"
      >
        <button
          className="bg-btn bg-btn--inactive"
          id="cancel"
          onClick={[Function]}
          type="button"
        >
          Cancel
        </button>
        <button
          className="bg-btn bg-btn--active"
          disabled={false}
          id="submit"
          type="submit"
        >
          Edit template
        </button>
      </div>
    </fieldset>
  </form>
</div>
`;
